iothub/device/src/Pipeline/MqttTransportHandler.cs

```diff
@@ -1054,14 +1080,29 @@ private Task HandleDisconnectionAsync(MqttClientDisconnectedEventArgs disconnect
             return Task.CompletedTask;
         }
 
+        private IncomingMessage ProcessC2DMessage(MqttApplicationMessageReceivedEventArgs receivedEventArgs)
+        {
+            byte[] payload = receivedEventArgs.ApplicationMessage.Payload;
+
+            var receivedCloudToDeviceMessage = new IncomingMessage(payload)
+            {
+                PayloadConvention = _payloadConvention,
+            };
+
+            PopulateMessagePropertiesFromMqttMessage(receivedCloudToDeviceMessage, receivedEventArgs.ApplicationMessage);
+
+            return receivedCloudToDeviceMessage;
+        }
+
         private async Task HandleReceivedMessageAsync(MqttApplicationMessageReceivedEventArgs receivedEventArgs)
         {
             receivedEventArgs.AutoAcknowledge = false;
             string topic = receivedEventArgs.ApplicationMessage.Topic;
 
             if (topic.StartsWith(_deviceBoundMessagesTopic, StringComparison.InvariantCulture))
             {
-                await HandleReceivedCloudToDeviceMessageAsync(receivedEventArgs).ConfigureAwait(false);
+                await HandleReceivedCloudToDeviceMessageAsync(ProcessC2DMessage(receivedEventArgs)).ConfigureAwait(false);
+                await receivedEventArgs.AcknowledgeAsync(CancellationToken.None).ConfigureAwait(false);
```

Comment:
Add some comments here explaining that messages are being added to a queue for later processing, and are being ack'ed immediately